Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunfactFragment.java

```diff
@@ -75,29 +76,48 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         if (fact != null) {
             (holder.desc).setText(fact.getText());
             (holder.title).setText(fact.getTitle());
+            if (fact.getSource() != null && !fact.getSource().equals("null")) {
+                (holder.text_source).setVisibility(View.VISIBLE);
+                (holder.source).setText(fact.getSource());
+                (holder.source).setOnClickListener(view1 -> {
+
+                    String url = fact.getSourceURL();
```

Comment:
move lines 84 to 93 in a new function (`openURL`), just like `shareImage` function.